cli/runtime/apis/config/v1alpha1/clientconfig_types.go

```diff
@@ -221,79 +221,83 @@ type OCIDiscovery struct {
-	GCPPluginRepository *GCPPluginRepository `json:"gcpPluginRepository,omitempty" yaml:"gcpPluginRepository"`
+	GCPPluginRepository *GCPPluginRepository `json:"gcpPluginRepository,omitempty" yaml:"gcpPluginRepository,omitempty"  mapstructure:"gcpPluginRepository,omitempty"`
 }
 
 // GCPPluginRepository is a plugin repository that utilizes GCP cloud storage.
 type GCPPluginRepository struct {
 	// Name of the repository.
-	Name string `json:"name,omitempty" yaml:"name"`
+	Name string `json:"name,omitempty" yaml:"name,omitempty" mapstructure:"name,omitempty"`
 
 	// BucketName is the name of the bucket.
-	BucketName string `json:"bucketName,omitempty" yaml:"bucketName"`
+	BucketName string `json:"bucketName,omitempty" yaml:"bucketName,omitempty" mapstructure:"bucketName,omitempty"`
 
 	// RootPath within the bucket.
-	RootPath string `json:"rootPath,omitempty" yaml:"rootPath"`
+	RootPath string `json:"rootPath,omitempty" yaml:"rootPath,omitempty" mapstructure:"rootPath,omitempty"`
 }
 
 // +kubebuilder:object:root=true
 
 // ClientConfig is the Schema for the configs API
 type ClientConfig struct {
-	metav1.TypeMeta   `json:",inline"`
-	metav1.ObjectMeta `json:"metadata,omitempty"`
-
+	metav1.TypeMeta   `json:",omitempty" yaml:",omitempty"`
+	metav1.ObjectMeta `json:",omitempty" yaml:",omitempty"`
 	// KnownServers available.
-	// Deprecation targeted for a a future version. Superseded by KnownContexts.
-	KnownServers []*Server `json:"servers,omitempty" yaml:"servers"`
+	// Deprecation targeted for a future version. Superseded by KnownContexts.
+	KnownServers []*Server `json:"servers,omitempty" yaml:"servers,omitempty" mapstructure:"servers,omitempty"`
 
 	// CurrentServer in use.
-	// Deprecation targeted for a a future version. Superseded by CurrentContext.
-	CurrentServer string `json:"current,omitempty" yaml:"current"`
+	// Deprecation targeted for a future version. Superseded by CurrentContext.
+	CurrentServer string `json:"current,omitempty" yaml:"current,omitempty" mapstructure:"current,omitempty"`
 
 	// KnownContexts available.
-	KnownContexts []*Context `json:"contexts,omitempty" yaml:"contexts"`
+	KnownContexts []*Context `json:"contexts,omitempty" yaml:"contexts,omitempty" mapstructure:"contexts,omitempty"`
 
 	// CurrentContext for every type.
-	CurrentContext map[ContextType]string `json:"currentContext,omitempty" yaml:"currentContext"`
+	CurrentContext map[ContextType]string `json:"currentContext,omitempty" yaml:"currentContext,omitempty" mapstructure:"currentContext,omitempty"`
 
 	// ClientOptions are client specific options.
-	ClientOptions *ClientOptions `json:"clientOptions,omitempty" yaml:"clientOptions"`
+	ClientOptions  *ClientOptions  `json:"clientOptions,omitempty" yaml:"clientOptions,omitempty" mapstructure:"clientOptions,omitempty"`
+	ConfigMetadata *ConfigMetadata `json:"configMetadata,omitempty" yaml:"configMetadata,omitempty" mapstructure:"configMetadata,omitempty"`
```

Comment:
Nit: This new section can use a comment given how generic sounding it is. 
Likewise some explanation of the PatchStrategy map would be helpful too.

Reply:
moved the configmetata to sep PR